### DISCUSSION TITLE
Data Persisting using Docker

### DISCUSSION THREAD

**Discussion by AshnaSabu123** [vanna-ai/vanna#602] [Q&A]
Hi Team,

We've used Docker to deploy the VANNA AI, but we'd need to know exactly how to persist the data using the volume, as well as how to train the data in both chroma and vector.

**Comment by erickrribeiro** [comment]
Could you adding more details, i.e Dockerfile or compose.yml file?

**Comment by AshnaSabu123** [comment]
Please have a check on the below docker-compose and Dockerfile that we have used for the deployment.


```version: '3.3'
services:
  vanna-ai:
    container_name: vanna-ai
    restart: always
    build:
      context: .
      dockerfile: Dockerfile-vanna-ai
    ports:
      - 5000:5000

Dockerfile

From python:3.10.12
RUN apt-get update && \
    apt-get install -y git
RUN git clone https://github.com/vanna-ai/vanna-flask.git
WORKDIR vanna-ai-flask
RUN ls -la
COPY .env .
RUN pip install -r requirements.txt
RUN pip install chromadb vanna[postgres] gevent
CMD ["python", "app.py"]
```

Please have a check on this and we are awaiting for the your response.

**Comment by rinshadka** [comment]
Thanks @erickrribeiro for the response. Actually we are thinking of an option to persist the training data even though ewe recreated the vanna-ai container, just like how its works for a postgres docker container. we are expecting there will be any option to declare docker volume to particular path in container for chroma and ventor databases. Thanks.

**Comment by rinshadka** [comment]
https://github.com/vanna-ai/vanna/discussions/402

https://github.com/vanna-ai/vanna/discussions/486

Linking  these related discussions here
